### DISCUSSION TITLE
Next Token Implementation Question

### DISCUSSION THREAD

**Discussion by marek715** [saleweaver/python-amazon-sp-api#230] [Q&A] [answered]
Hello!

When I try the following, I get an error that I need to provide CreatedAfter or LastUpdatedAfter:
`res = Orders(credentials=credentials).get_orders(next_token_param=next_token)`

Error:
`sp_api.base.exceptions.SellingApiBadRequestException: [{'code': 'InvalidInput', 'message': 'CreatedAfter or LastUpdatedAfter must be specified', 'details': ''}]`

But when I do provide those, it seems to return the same 100 orders over and over again (as if it ignores the "next_token_param" parameter.  

Below returns the same 100 items as if I did not have the "next_token_param" defined at all:
`res = Orders(credentials=credentials).get_orders(CreatedAfter=(datetime.now() - timedelta(days=cutoff_window)).isoformat(),next_token_param=next_token)`

I also attempted to use the load_all_pages decorator, but I was having trouble figuring out how to advance the generator.

Thank you,
Marek

**Comment by marek715** [comment]
1. (I did figure out how to use the iterator - for anyone asking the same question, answer below):
```
@load_all_pages()
def get_orders(**kwargs):
    return Orders(credentials=credentials).get_orders(**kwargs)

res=get_orders(CreatedAfter=(datetime.now() - timedelta(days=28)).isoformat())
for item in res:
     ProcessAmazonResults(item.payload)
```

2. Still not sure why this didn't work, but no longer needed because the above takes care of it much more cleanly
`res = Orders(credentials=credentials).get_orders(next_token_param=next_token)`

**Comment by saleweaver** [comment, marked as answer]
"next_token_param" is not the name of the key. Most likely it's "NextToken".

**Reply by marek715** [reply]
Thank you again, I thought I tried that earlier - but perhaps I tried an alternate spelling.  In either case, your implementation of load_all_pages is better than what I would do with NextToken.
